modules/xonotic/encryption.cpp

```diff
@@ -21,12 +21,19 @@
 
 #include <openssl/hmac.h>
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#define HMAC_CTX_init(ctx) ctx = HMAC_CTX_new()
+#define HMAC_CTX_cleanup HMAC_CTX_free
+#endif
+
+
 namespace xonotic {
 namespace crypto {
 
 std::string hmac_md4(const std::string& input, const std::string& key)
 {
-    HMAC_CTX *ctx = HMAC_CTX_new();
+    HMAC_CTX *ctx;
+    HMAC_CTX_init(ctx);
```

Comment:
I don't think a call to HMAC_CTX_init from an uninitialized pointer is valid.
The man page doesn't say much but from the sources is obvious that it must point to a valid object: http://docs.huihoo.com/doxygen/openssl/1.0.1c/hmac_8c_source.html#l00187

Reply:
According to the HMAC 1.1 docs, "HMAC_CTX_new() creates a new HMAC_CTX in heap memory" so the textually replaced version should be fine.  The 1.0 version is as it was in the original code but I haven't been able to wake up a Xonotic server host yet to confirm that the original worked.  1.0 docs say "HMAC_CTX_init() initialises a HMAC_CTX before first use".  Thanks for looking at this so soon.

Reply:
The problem is ctx is a pointer pointing to nothing

Reply:
Correct, I missed that.  Trying to keep the replacement clean but considering an #ifdef #else for the pointer assignment to stack allocated variable.

Reply:
To be honest at this point I think it's better to have the #ifs inside the function body rather than those weird bacros

Something like
```c++
#if 1.0
HMAC_CTX on_the_stack;
HMAC_CTX* ctx = &on_the_stack;
// initialize the 1.0 way
#else
HMAC_CTX* ctx = // initialize the 1.1 way
#endif

// ...

#if ...
// clean up
#endif
```